lisa/sut_orchestrator/qemu/platform.py

```diff
@@ -359,6 +359,24 @@ def _configure_nodes(self, environment: Environment, log: Logger) -> None:
                             yaml.safe_load(file)
                         )
 
+            # Configure data disks.
+            if node_space.disk:
+                assert isinstance(
+                    node_space.disk.data_disk_count, int
```

Comment:
I originally tried to use `node.capability` here. But the `data_disk_count` was an `IntRange` instead of an `int`. Is this expected behavior?

Reply:
The capability should be set by platform. Its default value is unlimited capacity from 0 to max.

The capability type can be rewritten by platform, like `AzureDiskOptionSettings`. But it needs to replace the default one by platform like `_convert_to_azure_node_space`. It's pretty complicated, and undocumented. Feel free to let me know, if there are more questions.

I'm planning to write some resource planner, it may be helpful for lab resource management. But don't rely on it, I don't have time allocated...